xacro/substitution_args.py

```diff
@@ -0,0 +1,576 @@
+#!/usr/bin/env python3
+
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Revision $Id: substitution_args.py 15178 2011-10-10 21:22:53Z kwc $
+
+"""
+Library for processing XML substitution args.
+
+This is currently used by roslaunch and xacro,
+but it is not yet a top-level ROS feature.
+This script has been taken from catkin's
+ros_comm/tools/roslaunch/src/roslaunch/substitution_args.py
+"""
+
+import math
+import os
+import sys
+
+import yaml
+
+from ament_index_python.packages import get_package_prefix
```

Comment:
Following a discussion on a separate thread (https://github.com/kazuki0824/xacro/pull/1#issuecomment-531821509), I think it's better to default to using `get_package_share_directory()`.

Reply:
I agree. It is in PR #212